### PR TITLE
[irods/irods_5183] 4.2.9 release notes

### PR DESCRIPTION
**Pull request by trel**

### CODE REVIEW

File: docs/release_notes.md [hunk 1]
```diff
@@ -1,3 +1,327 @@
...
+
+ - Replication to hierarchies [#3217] [#4686]
+
+ - Refactor iphymv [#3490] [#4212] [#4563] [#4896] [#5070] [#5177] [#5360] [#5446] [#5454]
+
+ - New msiTouch, itouch, and touch API plugin [#4669] [#4694] [#5152]
+
+ - Refactor checksum API [#3091] [#3282] [#3540] [#5251] [#5252] [#5263] [#5285] [#5288] [#5400] [#5401] [#5496]
+
+ - Add SHA1, SHA512, ADLER32, hashing functions [#3800] [#5392]
```
Comment (alanking) [inline docs/release_notes.md:21]: third comma is unnecessary

File: docs/release_notes.md [hunk 2]
```diff
@@ -1,3 +1,327 @@
...
+
+ - Add logical locking [#1781] [#3848] [#4170] [#4236] [#4433] [#4958] [#5312] [#5421] [#5486] [#5495] [#5542]
+
+ - Add client support for updating mtime of collections [#4190]
+
+ - New finalize_data_object API plugin [#2719] [#4331]
```
Comment (alanking) [inline docs/release_notes.md:27]: API is actually called data_object_finalize

File: docs/release_notes.md [hunk 3]
```diff
@@ -1,3 +1,327 @@
...
+
+ - Fix for --purgec option and bulk transfer [#3094]
+
+ - Document irm, irmtrash and physical directories [#3124]
+
+ - Fix for missing inline keyword [#3396]
+
+ - Fix for ifsck when it does not have enough permissions [#3428] [#5358]
+
+ - Fix for CMake IRODS_LINUX_DISTRIBUTION_VERSION_CODENAME[#3453]
```
Comment (alanking) [inline docs/release_notes.md:149]: missing space before [

File: docs/release_notes.md [hunk 4]
```diff
@@ -1,3 +1,327 @@
...
+
+ - Fix for irm with force flag [#5438]
+
+ - Fix for renaming local zone [#5445]
+
+ - Fix for msiRenameCollection to only rename collections [#5452]
+
+ - Fix for overwriting within special collections [#5457]
+
+ - Marked as wontfix or duplicate [#5472]
```
Comment (alanking) [inline docs/release_notes.md:291]: Should this be moved into and/or near the resolved/invalid list?

**Review by alanking — COMMENTED** [review]
Nitpicks, but looks good to me overall. Hitting approve if you wanna ignore my comments :)

EDIT: Well, I didn't hit approve because I missed the button lol I approve this change

File: docs/release_notes.md (hunk 4 above)
Comment (trel) [reply docs/release_notes.md:291]: sure, combined.

File: docs/release_notes.md (hunk 3 above)
Comment (trel) [reply docs/release_notes.md]: done

File: docs/release_notes.md (hunk 2 above)
Comment (trel) [reply docs/release_notes.md]: done

File: docs/release_notes.md (hunk 1 above)
Comment (trel) [reply docs/release_notes.md]: done

File: docs/release_notes.md [hunk 5]
```diff
@@ -1,3 +1,325 @@
...
+
+ - Tie query processor size to the future object [#4902]
+
+ - Unify storage resource voting [#4941]
+
+ - Allow query builder to clear specific query arguments [#4960]
+
+ - Support DATA_SIZE_KW in rsyncUtil and rcDataObjPut [#4987]
+
+ - Add istream support for target resource [#5000]
```
Comment (korydraughn) [inline docs/release_notes.md:47]: Maybe reword to something like:
```
istream can now target resources and replicas
```

File: docs/release_notes.md [hunk 6]
```diff
@@ -1,3 +1,325 @@
...
+
+ - Add RESOURCE_SKIP_VAULT_PATH_CHECK_ON_UNLINK resource property [#5030]
+
+ - Add IRODS_QUERY_ENABLE_SERVER_SIDE_API macro [#5033]
+
+ - Add IRODS_FOR_DOXYGEN macro [#5037]
+
+ - Add error() and state() system microservices [#5043]
+
+ - Add IRODS_FILESYSTEM_ENABLE_SERVER_SIDE_API macro [#5049]
```
Comment (korydraughn) [inline docs/release_notes.md:71]: This should say something like:
```
Improved client-side collection iterator performance
```

**Review by korydraughn — APPROVED** [review]
Looks good to me.

File: docs/release_notes.md (hunk 5 above)
Comment (trel) [reply docs/release_notes.md]: done

File: docs/release_notes.md (hunk 6 above)
Comment (trel) [reply docs/release_notes.md]: done

**Review by alanking — APPROVED** [review]
